llvm/include/llvm/IR/RepoTicket.h

```diff
@@ -116,98 +116,122 @@ class TicketNode : public MDNode {
   };
 
   TicketNode(LLVMContext &C, StorageType Storage,
-             GlobalValue::LinkageTypes Linkage, bool Pruned,
+             GlobalValue::LinkageTypes Linkage,
+             GlobalValue::VisibilityTypes Visibility, bool Pruned,
              ArrayRef<Metadata *> MDs)
       : MDNode(C, TicketNodeKind, Storage, MDs) {
     assert(MDs.size() == 2 && "Expected a hash and name.");
     static_assert(CheckLinkageType::isSafeCast(),
                   "Linkage type will overflow!");
-    SubclassData32 = static_cast<unsigned>(Linkage);
+    assert(static_cast<unsigned>(Linkage) < (1u << 16) &&
+           "Expected 16-bit linkage type");
+    assert(static_cast<unsigned>(Visibility) < (1u << 16) &&
+           "Expected 16-bit visibility type");
```

Comment:
> 
> 
> Should there be an equivalent of `CheckLinkageType` for `Visibility`? 
>
Good point. A template `CheckType` for `LInkage` and `Visibility` is added.

>We know that visibility can only have three values: since we’re bit-twiddling here anyway, could we limit this to 3 bits?
>
Yes, it is limited to 2 bits now.